### PR TITLE
Clarify li element role mapping

### PR DESCRIPTION
**Pull request by scottaohara**
The intent of this text change is to clarify that a list item that is still a child of a list, but has intervening generic elements should still be exposed as a listitem.

e.g.,

```
<li>with no list ancestor = generic</li>

<ul>
  <div> <!-- invalid html, but SHOULD be ignored since generic  -->
    <li> since this has 'accessibility parent' of list, it should remain exposed as a list item </li>
  </div>
</ul>
```

related to https://github.com/w3c/aria/issues/2137

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/pull/540.html" title="Last updated on Apr 18, 2024, 11:27 PM UTC (9cb316a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/540/f7e96ed...9cb316a.html" title="Last updated on Apr 18, 2024, 11:27 PM UTC (9cb316a)">Diff</a>

### CODE REVIEW

**Review by smhigley — APPROVED** [review]
simple and nice 👍

File: index.html [hunk 1]
```diff
@@ -5104,7 +5104,7 @@ <h4 id=el-li>`li`</h4>
     <tr>
       <th>Comments</th>
       <td>
-        If `li` element is not a child of <a data-cite="html">`ol`</a> , <a data-cite="html">`menu`</a> or <a data-cite="html">`ul`</a>, or if the containing
+        If `li` element is not an accessibility child of an <a data-cite="html">`ol`</a> , <a data-cite="html">`menu`</a> or <a data-cite="html">`ul`</a>, or if the containing
```
Comment (smhigley) [inline index.html]: only comment -- should this link to the ARIA spec def of accessibility child? (Guessing it should link to 7.3)

Comment (scottaohara) [reply index.html]: yes.  tbh i couldn't remember when i was writing this if that def existed yet

Comment (spectranaut) [reply index.html]: I think you can link with `[=accessibility child=]` per: https://respec.org/xref/?term=accessibility+child

```suggestion
        If `li` element is not an [=accessibility child=] of an <a data-cite="html">`ol`</a> , <a data-cite="html">`menu`</a> or <a data-cite="html">`ul`</a>, or if the containing
```

**Review by spectranaut — APPROVED** [review]
approve with link

**Review by cookiecrook — APPROVED** [review]

**Comment by scottaohara** [comment]
closing as this PR is moved over to the mono repo

**Pull request closed without merging**